tools/awslint/lib/rules/cloudwatch-events.ts

```diff
@@ -25,7 +25,7 @@ eventsLinter.add({
   message: `'onXxx()' methods should also be defined on construct interface`,
   eval: e => {
     for (const method of e.ctx.directEventMethods.concat(e.ctx.cloudTrailEventMethods)) {
-      e.assert(!e.ctx.interfaceType || e.ctx.interfaceType.allMethods.some(m => m.name === method.name), `${e.ctx.fqn}.${method.name}`);
+      e.assert(!e.ctx.interfaceType || e.ctx.interfaceType.allMethods.filter(m => !m.protected).some(m => m.name === method.name), `${e.ctx.fqn}.${method.name}`);
```

Comment:
@rix0rrr This is needed because the protected lifecycle methods in `constructs.Construct` are named `onSynthesize`, `onValidate` and `onPrepare`.

Reply:
Not forcing protected methods to be part of a public interface would always have been the right thing to do anyway, regardless of the change you just made.